packages/components/src/modal/index.tsx

```diff
@@ -71,16 +71,20 @@ function UnforwardedModal(
 	} = props;
 
 	const ref = useRef< HTMLDivElement >();
+
 	const instanceId = useInstanceId( Modal );
 	const headingId = title
 		? `components-modal-header-${ instanceId }`
 		: aria.labelledby;
-	const focusOnMountRef = useFocusOnMount( focusOnMount );
+	const focusOnMountRef = useFocusOnMount(
+		focusOnMount === 'firstContentElement' ? 'firstElement' : focusOnMount
+	);
```

Comment:
I suppose this conditional may not be necessary as the string with coerce to `true`. Does that alter the behaviour 🤔

Reply:
So this change means:

- since the hook signature is `useFocusOnMount( focusOnMount = 'firstElement' )` 
- focusOnMount absent from props it means `firstElement`, but if it's present and has a value of `firstContentElement` then it means `firstElement`.
- otherwise if it's present and has any other value nothing changes.

Reply:
As you can see the original code was:

```jsx
const focusOnMountRef = useFocusOnMount( focusOnMount );
```

The new code means that if `firstContentElement` is passed as the value of the _Modal's_ `focusOnMount` _prop_, then we force the hook to focus the "first element" in the element to which the `ref` returned from the hook is applied. 

The default for the hook is `firstElement` but that could change and so I prefer to be absolutely explicit in the behaviour we want and expect rather than rely on defaults. For this reason I map `firstContentElement` to `firstElement`.

You will not that in the Modal's rendering code, we conditionally apply the `ref` returned from the `focusOnMount` hook to different target JSX elements based on the value of the Modal's `focusOnMount` _prop_. If `firstContentElement` is passed then the consumer wishes to place focus on the first element within the Modal's _content_ and thus we apply the `ref` to that element. Otherwise we default to previous behaviour.

I hope that helps to explain this unusual code? I have also provided a comment to further enhance this.

Reply:
This is the gist of the explanation:

> The default for the hook is firstElement but that could change and so I prefer to be absolutely explicit in the behaviour we want and expect rather than rely on defaults. For this reason I map firstContentElement to firstElement.

Thank you!